src/fastgltf.cpp

```diff
@@ -1303,366 +1313,445 @@ fg::Expected<fg::Asset> fg::Parser::parse(simdjson::dom::object root, Category c
+                        if (accessor.componentType != ComponentType::Float && accessor.componentType != ComponentType::Double) {
+                            return Error::InvalidGltf;
                         }
-                        case dom::element_type::INT64: {
-                            std::int64_t value;
-                            if (element.get_int64().get(value) != SUCCESS) {
-                                return Error::InvalidGltf;
-                            }
 
-							if (auto* doubles = std::get_if<double_vec>(&variant); doubles != nullptr) {
-								(*doubles)[idx++] = static_cast<double>(value);
-							} else if (auto* ints = std::get_if<int64_vec>(&variant); ints != nullptr) {
-								(*ints)[idx++] = static_cast<std::int64_t>(value);
-							} else {
-								return Error::InvalidGltf;
-							}
-                            break;
+                        if (!std::holds_alternative<double_vec>(variant)) {
+                            return Error::InvalidGltf;
                         }
-                        case dom::element_type::UINT64: {
-                            // Note that the glTF spec doesn't care about any integer larger than 32-bits, so
-                            // truncating uint64 to int64 wouldn't make any difference, as those large values
-                            // aren't allowed anyway.
-                            std::uint64_t value;
-                            if (element.get_uint64().get(value) != SUCCESS) {
-                                return Error::InvalidGltf;
-                            }
-
-							if (auto* doubles = std::get_if<double_vec>(&variant); doubles != nullptr) {
-								(*doubles)[idx++] = static_cast<double>(value);
-							} else if (auto* ints = std::get_if<int64_vec>(&variant); ints != nullptr) {
-								(*ints)[idx++] = static_cast<std::int64_t>(value);
-							} else {
-								return Error::InvalidGltf;
-							}
-                            break;
+                        std::get<double_vec>(variant).emplace_back(num.get_double());
+                        break;
+                    }
+                    case ondemand::number_type::signed_integer: {
+                        std::int64_t value = num.get_int64();
+
+                        if (auto* doubles = std::get_if<double_vec>(&variant); doubles != nullptr) {
+                            (*doubles).emplace_back(static_cast<double>(value));
+                        } else if (auto* ints = std::get_if<int64_vec>(&variant); ints != nullptr) {
+                            (*ints).emplace_back(static_cast<std::int64_t>(value));
+                        } else {
+                            return Error::InvalidGltf;
+                        }
+                        break;
+                    }
+                    case ondemand::number_type::unsigned_integer: {
+                        // Note that the glTF spec doesn't care about any integer larger than 32-bits, so
```

Comment:
It should only trigger `unsigned_integer` if the integer is larger or equal to 2**63, so maybe this case is in error for you, or needs to be treated as a double?

Reply:
The data included in the arrays can at most be 2\*\*32 (or 32-bit floats), so that that case is just never reached probably. If it's only ever taken with values larger or equal to 2**63 I guess I would be safe to just remove it. Though for correctness, would it really be bad to keep it?

Reply:
> would it really be bad to keep it?

I recommend that you simplify your code prior to optimizing it. It is much easier to optimize simpler code.

I tried dumping the assembly from this function and I got several pages of instructions. It is just not possible to understand that much code at once. If you can trim it down to something simpler, then it might be much easier to understand where the performance challenges are.